beetsplug/fetchart.py

```diff
@@ -115,75 +105,64 @@ def _validate(self, plugin):
                     self._log.debug('image is not close enough to being '
                                     'square, ({} - {} > {})',
                                     long_edge, short_edge, plugin.margin_px)
-                    return self.CANDIDATE_BAD
+                    return False
             elif plugin.margin_percent:
                 margin_px = plugin.margin_percent * long_edge
                 if edge_diff > margin_px:
                     self._log.debug('image is not close enough to being '
                                     'square, ({} - {} > {})',
                                     long_edge, short_edge, margin_px)
-                    return self.CANDIDATE_BAD
+                    return False
             elif edge_diff:
                 # also reached for margin_px == 0 and margin_percent == 0.0
                 self._log.debug('image is not square ({} != {})',
                                 self.size[0], self.size[1])
-                return self.CANDIDATE_BAD
+                return False
 
         # Check maximum dimension.
-        downscale = False
         if plugin.maxwidth and self.size[0] > plugin.maxwidth:
             self._log.debug('image needs rescaling ({} > {})',
                             self.size[0], plugin.maxwidth)
-            downscale = True
+            self.downscale = True
 
         # Check filesize.
-        downsize = False
         if plugin.max_filesize:
             filesize = os.stat(syspath(self.path)).st_size
             if filesize > plugin.max_filesize:
                 self._log.debug('image needs resizing ({}B > {}B)',
                                 filesize, plugin.max_filesize)
-                downsize = True
+                self.downsize = True
 
         # Check image format
-        reformat = False
         if plugin.cover_format:
             fmt = ArtResizer.shared.get_format(self.path)
-            reformat = fmt != plugin.cover_format
-            if reformat:
+            self.reformat = fmt != plugin.cover_format
+            if self.reformat:
                 self._log.debug('image needs reformatting: {} -> {}',
                                 fmt, plugin.cover_format)
 
-        if downscale:
-            return self.CANDIDATE_DOWNSCALE
-        elif downsize:
-            return self.CANDIDATE_DOWNSIZE
-        elif plugin.deinterlace:
-            return self.CANDIDATE_DEINTERLACE
-        elif reformat:
-            return self.CANDIDATE_REFORMAT
-        else:
-            return self.CANDIDATE_EXACT
+        return True
 
     def validate(self, plugin):
         self.check = self._validate(plugin)
         return self.check
 
-    def resize(self, plugin):
-        if self.check == self.CANDIDATE_DOWNSCALE:
+    def adjust(self, plugin):
```

Comment:
Similarly, it would be really useful to add a docstring here that says that this method is sensitive to the various flags on the `Candidate` object.